File: ts/128/121.ts
function maxProfit(prices: number[]): number {
  let profit = 0, minPrice = prices[0]
  for (const v of prices) {
    if (v < minPrice) {
      minPrice = v
    } else {
      profit = Math.max(profit, v - minPrice)
    }
  }
  return profit
};

File: c/128/121.c
#include <limits.h>

int maxProfit(int* prices, int pricesSize){
  int profit = 0, min = INT_MAX;
  for (int i = 0; i < pricesSize; ++i) {
    if (prices[i] > min) {
      int diff = prices[i] - min;
      profit = diff > profit ? diff : profit;
    } else if (prices[i] < min) {
      min = prices[i];
    }
  }
  return profit;
}